quadratichq/quadratic
# feat: preserve `file` query param through auth0 process

Today, a logged in user can add a `file` query param that will import a remote `.grid` file into their version of the app (and change the url from `?file=...` to `?local=...`

However, if the user is not logged in, that file param gets lost in through the auth0 redirects. So we fix that by seeing if the `file` param is present when the user is not logged in. If it is, we store it in sessionStorage, then retrieve it (and delete it) when they get redirected back to the app from Auth0.

To test:

- Make sure you are logged out of the app
- Visit the preview branch with a `file` query param, e.g. https://pr-412.de5w5iglj13on.amplifyapp.com/?file=https://cdn.jim-nielsen.com/files/sample-file.grid
- Log in and see that the above file is imported after login
